src/web/rustdoc.rs

```diff
@@ -448,7 +448,19 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
     if !path.ends_with(".html") {
         rendering_time.step("serve asset");
 
-        return Ok(File(blob).serve());
+        let mut response = File(blob).serve();
+        // default asset caching behaviour is `Cache::ForeverInCdnAndBrowser`.
+        // We don't want to cache invocation specific assets in the browser
+        // but only in the CDN so we can invalidate these after a build.
```

Comment:
The double negative between "don't want" and "but only" is confusing me a bit. How about:

"We want invocation specific assets cached in the CDN but not in the browser. That way we can invalidate the CDN cache after a build."

Reply:
changed. 

This is also a topic that might be worth discussing. 

While this change would resolve #1593, it actually could make search _slower_ since we don't allow to locally the search index any more.

Reply:
Yeah, that is a bummer. A few ways forward:

 1. We can add code that says "invocation specific _JS_ assets are infinitely cached." This would preserve today's behavior (#1593), which is somewhat wrong but not causing massive problems.
 2. We could add ETags to invocation specific JS assets, so the CDN can at least reply 304 Not Modified.
 3. We could deploy this, take the speed hit for now, and wait for follow-on work on the rustdoc side that will fix the issue of invocation-specific JS assets having colliding filenames.
 
My preference is (1). Reasoning: When making big changes / refactorings, it's good not to pull in incidental things if possible. Yes, the current behavior is wrong, but we should fix it as a separate step.

Reply:
> When making big changes / refactorings, it's good not to pull in incidental things if possible. 

I'm with you here. 

I reverted the cache settings now, and we keep solving this separately. 

Option 4 could also be to use the `ForeverInCdnAndStaleInBrowser` policy, ETags are useful in any case for revalidating content.